src/plugins/intel_cpu/src/transformations/cpu_opset/common/op/read_value_with_subgraph.hpp

```diff
@@ -0,0 +1,45 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/op/op.hpp"
+#include "openvino/op/read_value.hpp"
+#include "openvino/op/util/sub_graph_base.hpp"
+#include "transformations/cpu_opset/common/op/submodel.hpp"
+
+namespace ov {
+namespace intel_cpu {
+
+class ReadValueWithSubgraphNode : public op::util::MultiSubGraphOp {
+public:
+    OPENVINO_OP("ReadValueWithSubgraph", "cpu_plugin_opset");
+
+    ReadValueWithSubgraphNode();
+    ReadValueWithSubgraphNode(const std::shared_ptr<ov::op::util::Variable>& variable);
+
+    std::string get_variable_id() const;
+
+    void set_body(const std::shared_ptr<Model>& body) {
+        m_bodies[0] = body;
+    }
+    const std::shared_ptr<Model>& get_body() const {
+        return m_bodies[0];
+    }
+
+    void set_input(const Output<Node>& value, const std::shared_ptr<op::v0::Parameter>& body_parameter);
+
+    Output<Node> set_output(const std::shared_ptr<op::v0::Result>& body_result);
```

Comment:
Do we really need those setters?
Can't we just pass ov::Model to the constructor, as it is done for SubModel operation?

Reply:
It is not necessary, we can call base class function, but I think it has better readability. For example: [void If::set_input](https://github.com/xipingyan/openvino/blob/79b82728e583cfd42d99da97904a3a804d908672/src/core/include/openvino/op/if.hpp#L68-L78)

Reply:
Those setters are more like workarounds rather than a good implementation example.
If an object can be constructed within a constructor, it should.
If we do not need setters, we should not introduce them.

Reply:
Hi @EgorDuplensky , As you said, I tried to move them to constructor, but failed.
If I moved them, [shared_from_this](https://github.com/xipingyan/openvino/blob/9c0989da2e4bb07d3fd55a6644cd50622d935438/src/core/src/op/util/multi_subgraph_base.cpp#L142) would be called in constructor, and then throwed `nullptr`, because constructor must finish, `shared_from_this()` can return correct value.

Reply:
Could you please clarify, from where `shared_from_this` is being called in this case?

Reply:
Move to Constructor, I also need to move [`set_body_outputs`](https://github.com/xipingyan/openvino/blob/9c0989da2e4bb07d3fd55a6644cd50622d935438/src/core/src/op/util/multi_subgraph_base.cpp#L131C53-L131C69) to Constructor, `shared_from_this()` will be called in the function: `set_body_outputs`